src/joystick/SDL_gamepad.c

```diff
@@ -598,20 +607,20 @@ static GamepadMapping_t *SDL_CreateMappingForAndroidGamepad(SDL_JoystickGUID gui
 
     SDL_strlcpy(mapping_string, "none,*,", sizeof(mapping_string));
 
-    if (button_mask & (1 << SDL_GAMEPAD_BUTTON_A)) {
+    if (button_mask & (1 << SDL_GAMEPAD_BUTTON_SOUTH)) {
         SDL_strlcat(mapping_string, "a:b0,", sizeof(mapping_string));
```

Comment:
This probably deserves a comment about how, for historical reasons, the mapping says `a` when it means "south" and so on.

Do all gamepads that will go through this function have a = south, or will some of them have the Nintendo-style layout where SDL toggles between position-based and label-based mappings according to a hint?

Reply:
This is a good question - I would definitely be up for deprecating a/b/x/y in favor of n/s/e/w for mappings; we could still have something like USE_LABELS as a config value to indicate devices that prefer bayx or sony layouts.

Reply:
I believe during previous discussion @slouken's position was that we can't deprecate a/b/x/y in the serialized string form of mappings any time soon, because projects like https://github.com/gabomdq/SDL_GameControllerDB will want to share a single table of mappings between SDL 2 and SDL 3 games.

Reply:
It might be good to do something like this in the string form, though:

* when parsing the string: accept either n/s/e/w or a/b/x/y, ideally treating n/s/e/w as "more important" and falling back to a/b/x/y if absent
* when generating the string: duplicate n/s/e/w as a/b/x/y for backwards compat with SDL 2